### DISCUSSION TITLE
OpenCopilot does not understand which site it is executing on

### DISCUSSION THREAD

**Discussion by VadymPogorelov** [opencx-labs/copilot#92] [General]
Perhaps I am writing in the wrong place - in that case please give me the correct contacts.

I am trying to install CoPilot on a test site.
I have performed the following steps:
1) Prepared the endpoint API to change the site logo
2) Created swagger-file with API description
3) Got the code to insert into my site https://prnt.sc/ypIeRYsZv8iS.
4) I inserted it, replacing the necessary fragments (button id, welcome message, access token to my API) https://prnt.sc/qqV_40qTX5Ue.

But, as you can see on the last screen, Copilot thinks it is running on www.example.com.

Accordingly, it cannot access the API and replace the logo on the test site https://trex3.themerex.net using the description from the swagger file.

Is it possible that I am doing something wrong?

**Comment by codebanesr** [comment]
update the servers section in your swagger file, it should work

**Comment by VadymPogorelov** [comment]
What's wrong with the servers section? 
https://prnt.sc/Tx-UAIxZdgAy

On the left are the settings of the 'servers' section, on the right is the result of a test GET request directly in the Swagger-editor. The result is correct, it returned the correct URL of the current logo.
Next, make a PUT request to change the logo directly in Swagger-editor: https://prnt.sc/Zo_Vh4vI1OAq.
New logo installed https://prnt.sc/dq9XVG_WX6HX
As you can see - everything is OK when accessing the API directly from Swagger-editor.

But when I install the generated code fragment https://prnt.sc/lWA3VoDooLuy on my test site https://prnt.sc/WfZTpYD7GXvb (the one specified in the servers section of the swagger-file), nothing happens (there are not even any calls to the server in the logs).
And as I wrote earlier, when asked "Show the URL of the current site" it answers https://www.example.com. 
https://prnt.sc/xQ-mrHB8S07k
How can it change something on https://trex3.themerex.net if it doesn't even understand what is being executed on this site?

**Reply by codebanesr** [reply]
Hey @VadymPogorelov ,
I see what you are saying now.

We are working on serp integration that will allow the agent to look for valid images on the internet. Since the product is still in beta testing, there are some features that aren't fully implemented yet. Please be patient as we work to improve the system. I have documented your requirements so we will prioritize them in the very next release.

**Comment by VadymPogorelov** [comment]
Thanks for the quick reply.

It's not that it doesn't access the picture I'm asking it to set as a logo. It doesn't need to, my code downloads it. Copilot's task is simply to call the API and pass it the URL of the image in the parameters (as in the editor tests in the screens above). 
If I understand the order of its operation correctly, then when the script is initialised it is passed a token, it contacts your server, this token determines which Copilot is used and the Swagger file attached to it should be used to call the corresponding endpoint from my API.
But nothing happens.

**Comment by VadymPogorelov** [comment]
One more thing - when Copilot is running, js errors keep appearing in the console https://prnt.sc/ncoBdbxShHQc

**Comment by codebanesr** [comment]
@VadymPogorelov This issue is being tracked [here](https://github.com/openchatai/OpenCopilot/issues/101). I'm also working on a new swagger parser, and another colleague is working on a flows editor. Hopefully, these issues will go away on their own or we will be able to fine-tune the llm server.
